Foundation/NSTask.swift

```diff
@@ -24,80 +24,82 @@ private func WEXITSTATUS(status: CInt) -> CInt {
-            
-            if task.terminationHandler != nil {
-                var threadID = pthread_t()
-                pthread_create(&threadID, nil, { (context) -> UnsafeMutablePointer<Void> in
-                    
-                    let unmanagedTask : Unmanaged<NSTask> = Unmanaged.fromOpaque(context)
-                    let task = unmanagedTask.takeRetainedValue()
-                    
-                    task.terminationHandler!( task )
-                    return context
-                    
-                    }, UnsafeMutablePointer<Void>(Unmanaged.passRetained(task).toOpaque()))
-            }
-            
-            // Set the running flag to false
-            
-            task.running = false
-            
-            // Invalidate the source and wake up the run loop, if it's available
-            
-            CFRunLoopSourceInvalidate(task.runLoopSource)
-            if let runLoop = task.runLoop {
-                CFRunLoopWakeUp(runLoop._cfRunLoop)
-            }
-        }
-        
-        // Clear out the NSTask from the process table
-        
-        processMap[pid] = nil
-        
-        processLaunchLock.unlock()
+    let unmanagedrunLoopA = Unmanaged<AnyObject>.fromOpaque(COpaquePointer(a))
+    guard let runLoopA = unmanagedrunLoopA.takeUnretainedValue() as? NSRunLoop else {
+        return false
     }
+    
+    let unmanagedRunLoopB = Unmanaged<AnyObject>.fromOpaque(COpaquePointer(a))
+    guard let runLoopB = unmanagedRunLoopB.takeUnretainedValue() as? NSRunLoop else {
+        return false
+    }
+    
+    guard runLoopA == runLoopB else {
+        return false
+    }
+    
+    return true
 }
 
-private func runLoopCallback(context : UnsafeMutablePointer<Void>) -> Void {}
+@noreturn private func managerThread(x: UnsafeMutablePointer<Void>) -> UnsafeMutablePointer<Void> {
+    
+    managerThreadRunLoop = NSRunLoop.currentRunLoop()
+    var emptySourceContext = CFRunLoopSourceContext (version: 0, info: UnsafeMutablePointer<Void>(Unmanaged.passUnretained(managerThreadRunLoop!).toOpaque()),
+                                                              retain: runLoopSourceRetain, release: runLoopSourceRelease, copyDescription: nil,
+                                                                      equal: runloopIsEqual, hash: nil, schedule: nil, cancel: nil,
+                                                                             perform: emptyRunLoopCallback)
+    
+    CFRunLoopAddSource(managerThreadRunLoop?._cfRunLoop, CFRunLoopSourceCreate(kCFAllocatorDefault, 0, &emptySourceContext), kCFRunLoopDefaultMode)
+    
+    managerThreadRunLoopIsRunningCondition.lock()
+    
+    CFRunLoopPerformBlock(managerThreadRunLoop?._cfRunLoop, kCFRunLoopDefaultMode) { 
+        managerThreadRunLoopIsRunning = true
+        managerThreadRunLoopIsRunningCondition.broadcast()
+        managerThreadRunLoopIsRunningCondition.unlock()
+    }
+    
+    managerThreadRunLoop?.run()
+    fatalError()
```

Comment:
probably should put a message here of why it is an error to exit the run loop; future selves will appreciate it